crowd_anki/export/note_sorter.py

```diff
@@ -0,0 +1,60 @@
+from dataclasses import dataclass
+
+from ..config.config_settings import ConfigSettings
+
+DEBUG = False
+
+@dataclass
+class NoteSorter():
+    sort_method: list
+    is_reversed: bool
+    skip_sorting: bool
+    sort_key_tuple: tuple
+
+    @classmethod
+    def from_config(cls, config: ConfigSettings):
+        sorting_definitions = {
+            ConfigSettings.DeckExportSortMethods.NO_SORTING: None,
+            ConfigSettings.DeckExportSortMethods.GUID: lambda i: i.anki_object.guid,
+            ConfigSettings.DeckExportSortMethods.FLAG: lambda i: i.anki_object.flags,
+            ConfigSettings.DeckExportSortMethods.TAG: lambda i: i.anki_object.tags,
+            ConfigSettings.DeckExportSortMethods.NOTEMODEL: lambda i: i.anki_object._model["name"],
+            ConfigSettings.DeckExportSortMethods.NOTEMODELID: lambda i: i.anki_object._model["crowdanki_uuid"],
+            ConfigSettings.DeckExportSortMethods.FIELD1: lambda i: i.anki_object.fields[0],
+            ConfigSettings.DeckExportSortMethods.FIELD2: lambda i: i.anki_object.fields[1]
+        }
+        
+        cls.sort_method = [ConfigSettings.DeckExportSortMethods._value2member_map_[method] for method in config.export_deck_sort_methods]
+        cls.is_reversed = config.export_deck_sort_reversed
```

Comment:
unless I'm missing something I don't believe you want to assign anything to the class instance

Reply:
sure, you can, but why would you want to in that case?